python/ray/data/dataset.py

```diff
@@ -1056,6 +1069,132 @@ def aggregate(self, *aggs: AggregateFn) -> U:
         ret = self.groupby(None).aggregate(*aggs).take(1)
         return ret[0] if len(ret) > 0 else None
 
+    def aggregate_vectorized(self, *aggs: VectorizedAggregation) -> U:
```

Comment:
Should be combined with the existing aggregate() (support both AggregateFn and VectorizedAggregateFn).